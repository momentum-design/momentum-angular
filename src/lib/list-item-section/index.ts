export * from './list-item-section.module';
export * from './list-item-section.component';
