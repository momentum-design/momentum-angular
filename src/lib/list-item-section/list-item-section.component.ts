/** @component list-item-section */
import { Component, OnInit, Input, HostBinding } from '@angular/core';

@Component({
  selector: 'md-list-item-section',
  template: `
    <ng-content></ng-content>
  `,
  styles: [],
})
export class ListItemSectionComponent implements OnInit {
  constructor() {}

  /** @option position Determine the ListItemSection's position | 'center' */
  @Input() position = 'center';
  /** @option class Optional css class name | '' */
  @Input() class = '';

  @HostBinding('class') get className(): string {
    return (
      `md-list-item__${this.position}` +
      `${(this.class && ` ${this.class}`) || ''}`
    );
  }

  ngOnInit() {
    if (this.position && !this.isValidPosition()) {
      throw new Error(`md-list-item-section: ListItemSection position option
        must be one of the following: left, center, right, center-align`);
    }
  }

  private isValidPosition = () =>
    ['left', 'center', 'right', 'center-align'].includes(this.position)
}
