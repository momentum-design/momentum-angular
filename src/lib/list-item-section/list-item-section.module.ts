import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListItemSectionComponent } from './list-item-section.component';

@NgModule({
  imports: [
    CommonModule
  ],
  declarations: [ListItemSectionComponent],
  exports: [ListItemSectionComponent]
})
export class ListItemSectionModule { }
