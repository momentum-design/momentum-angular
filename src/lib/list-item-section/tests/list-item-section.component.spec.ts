import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ListItemSectionComponent } from '../list-item-section.component';

describe('ListItemSectionComponent', () => {
  let component: ListItemSectionComponent;
  let fixture: ComponentFixture<ListItemSectionComponent>;
  let nativeElement: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ListItemSectionComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ListItemSectionComponent);
    component = fixture.componentInstance;
  });

  it('should create and match snapshot', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should modify class with additional classes', () => {
    component.class = 'myNewClass';
    fixture.detectChanges();

    nativeElement = fixture.nativeElement;
    expect(nativeElement.className).toEqual('md-list-item__center myNewClass');
  });

  it('should modify class when position option is defined', () => {
    component.position = 'left';
    fixture.detectChanges();

    nativeElement = fixture.nativeElement;
    expect(nativeElement.className).toEqual('md-list-item__left');
  });
});
