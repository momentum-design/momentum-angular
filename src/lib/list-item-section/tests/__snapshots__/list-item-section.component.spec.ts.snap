// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemSectionComponent should create and match snapshot 1`] = `
<md-list-item-section
  class=""
  isValidPosition={[Function Function]}
  position={[Function String]}
>

</md-list-item-section>
`;
